### DISCUSSION TITLE
About next(model.parameters())

### DISCUSSION THREAD

**Discussion by ChunZhuo** [mrdbourke/pytorch-deep-learning#439] [General]
torch.manual_seed(42)
model_1 = LinearRegressionModel()
model_1, model_1.state_dict()

OUTPUT:
(LinearRegressionModel(),
 OrderedDict([('weights', tensor([0.1940], device='cuda:0')),
              ('bias', tensor([0.1391], device='cuda:0'))]))

next(model_1.parameters()).device
p1 = next(model_1.parameters())
p2 = next(model_1.parameters())
print(f"There are the params {p1}, {p2}")
print(type(model_1.parameters()))

OUTPUT:
There are the params Parameter containing:
tensor([0.1940], device='cuda:0', requires_grad=True), Parameter containing:
tensor([0.1940], device='cuda:0', requires_grad=True)
<class 'generator'>

Why I cannot get bias using p2 = next(model_1.parameters())?
next function can be used for every object class in pytorch?

**Comment by adamomeister** [comment]
Are you trying to view your model's weights and biases? If so, I think you need to use the `state_dict()` method.

**Comment by ChunZhuo** [comment]
Yes, using state_dict() can work. I also do not understand that next() function can iterate over all the parameters, but why I can only get the weights? I think if I use next() method twice, I can get weights and bias also.
________________________________
